nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/HealthCheckService.java

```diff
@@ -164,21 +168,50 @@ public Object run(final HealthCheckServiceConfiguration configurationProxy)
             String historicalTraceStoreSizeConfig = configuration.getHistoricalTraceStoreSize();
             if (historicalTraceStoreSizeConfig != null) {
                 this.historicalTraceStoreSize = Integer.parseInt(historicalTraceStoreSizeConfig);
+
+                String historicalTraceStoreTimeLimit = configuration.getHistoricalTraceStoreTimeout();
+                if (historicalTraceStoreTimeLimit != null) {
+                    this.historicalTraceStoreTimeout = TimeUtil.setStoreTimeLimit(historicalTraceStoreTimeLimit);
+                }
             }
         }
     }
 
     public void bootstrapHealthCheck() {
         if (configuration != null) {
+            final Thread.UncaughtExceptionHandler exceptionHandler = new Thread.UncaughtExceptionHandler() {
+                public void uncaughtException(Thread th, Throwable ex) {
+                    logger.log(Level.SEVERE, "Uncaught exception in Health Check thread " + ex);
+                }
+            };
+
             executor = Executors.newScheduledThreadPool(configuration.getCheckerList().size(),  new ThreadFactory() {
                 public Thread newThread(Runnable r) {
-                    return new Thread(r, PREFIX + threadNumber.getAndIncrement());
+                    Thread thread = new Thread(r, PREFIX + threadNumber.getAndIncrement());
+                    thread.setUncaughtExceptionHandler(exceptionHandler);
+                    return thread;
                 }
             });
+
+            historicCleanerExecutor = Executors.newScheduledThreadPool(1,  new ThreadFactory() {
```

Comment:
one is dealing with the checkers and the other one w/ just one cleanup task. in order not to break the generic approach on checkers, that is having an arbitrary number of health check mechanisms defined, I used a separate pool.